enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -0,0 +1,434 @@
+spec:
+  selector:                                                         # (1)
+    istio: ingressgateway
+  gatewayClassName: istio                                           # (2)
+  addresses:
+  - value: istio-ingressgateway.openshift-ingress.svc.cluster.local # (3)
+    type: Hostname
+  listeners:
+  - name: http
+    hostname: "*.gwapi.example.com"
+    port: 80
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: All
+  - name: https
+    hostname: "*.gwapi.example.com"
+    port: 443
+    protocol: HTTPS
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - name: gwapi-wildcard
+    allowedRoutes:
+      namespaces:
+        from: All
+```
+[1]: The requirement to specify `selector.istio` is an Istio-ism that needs to be investigated. \
+[2]: The requirement to specify `gatewayClassName: istio ` is another Istio-ism that is addressed by \
+     [istio/istio#38134](https://github.com/istio/istio/issues/38134) and \
+     [istio/istio#41884](https://github.com/istio/istio/pull/41884). \
+[3]: The requirement to specify the "istio-ingressgateway" service's DNS name \
+     in `spec.addresses` is another Istio-ism that needs to be discussed.
+
+Then a project admin can attach an HTTPRoute to the Gateway.
+
+_"As a cluster admin, I want to configure ingress using Gateway API, and I want
+to enable application owners to configure mesh using the Istio API, and I want
+these two things to work harmoniously."_
+
+The goals of this enhancement include that ingress and mesh use a unified
+control-plane (with a single Istiod instance).  The cluster admin should be able
+to enable mesh and subsequently enable ingress or vice versa without needing to
+reconfigure the mesh or ingress, respectively, in the subsequent step.
+
+_"As a project admin, I want to configure ingress to my application using an
+HTTPRoute, the same way I would use an OpenShift Route."_
+
+Once the cluster admin has configured a Gateway as described in the earlier user
+story, the project admin can define an HTTPRoute that attaches to the Gateway.
+For example:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: http
+  namespace: example-app
+spec:
+  parentRefs:
+  - name: gateway
+    namespace: openshift-ingress
+  hostnames: ["test.gwapi.example.com"]
+  rules:
+  - backendRefs:
+    - name: example-app
+      port: 8080
+```
+
+This example exposes the "example-app" service, using port 8080 of the service.
+In this example, the HTTPRoute specifies a host name that matches both listeners
+on the Gateway, so the application is exposed using both HTTP and HTTPS.  When a
+client connects to the application using HTTPS, the default certificate that the
+cluster admin specified on the Gateway is used.  The Envoy proxy terminates the
+TLS connection and forwards the HTTP connection to the "example-app" service.
+
+### Goals
+
+* Automate the installation of Istio/Envoy by means of OpenShift Service Mesh.
```

Comment:
```suggestion
* Automate the installation of Istio/Envoy as a Red Hat-supported solution by means of OpenShift Service Mesh.
```

Reply:
Fair point.  I have changed it to "a Gateway API implementation for OpenShift based on Istio/Envoy".